flake.nix

```diff
@@ -109,25 +102,17 @@
         { rustProfile ? "minimal"
         , rustExtensions ? [
             "rust-src"
-            "rust-analysis"
+            "rust-analyzer"
```

Comment:
I wonder if rust-analyzer should be here, and propagated to all builds. Shouldn't we only include it in the mkShell? Or is it better for caching purposes to just have one global setting and use the same setup everywhere?

Reply:
Good point, I moved it (and rust-src) to the dev shell. It shouldn't affect caching, as the dev shell already has a slightly different toolchain (e.g. with a wasm32 target).